File: server/src/main/java/com/uiptv/util/MacAttributeParser.java
package com.uiptv.util;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class MacAttributeParser implements StalkerAttributeParser {
    private static final Pattern MAC_PATTERN = Pattern.compile("(?i)(([0-9A-F]{2}[:-]){5}([0-9A-F]{2}))");

    @Override
    public String parse(String line) {
        Matcher m = MAC_PATTERN.matcher(line);
        return m.find() ? m.group(1) : null;
    }

    @Override
    public StalkerAttributeType getAttributeType() {
        return StalkerAttributeType.MAC;
    }
}

File: server/src/main/kotlin/com/uiptv/util/MacAttributeParser.kt
package com.uiptv.util

import com.uiptv.api.StalkerAttributeParser
import java.util.regex.Pattern

class MacAttributeParser : StalkerAttributeParser {
    override fun parse(line: String): String? {
        val matcher = MAC_PATTERN.matcher(line)
        return if (matcher.find()) matcher.group(1) else null
    }

    override fun getAttributeType(): StalkerAttributeType = StalkerAttributeType.MAC

    private companion object {
        val MAC_PATTERN: Pattern = Pattern.compile("(?i)(([0-9A-F]{2}[:-]){5}([0-9A-F]{2}))")
    }
}
